### ISSUE TITLE
DiRT 4 (421020)

### ISSUE THREAD

**Issue by mimattr** [ValveSoftware/Proton#992]
Ubuntu 16.04, NVIDIA GTX 970, Proton 3.7-4 Beta:

[Steam System Information](https://gist.github.com/mimattr/b0b7f46366e31e254460b9593ce87e51)

Test | Result
-- | --
Singleplayer | Working as expected^
Local Co-op | _N/A_
Online Multiplayer | Working as expected^
VR | _NOT TESTED_

Configuration | ...
-- | --
Input | Steam Controller
Display | 1920x1080
Fullscreen | Yes
Preset | Medium^^
VSync | Off
API | DX11

^ Unable to connect to [RaceNet](https://racenet.codemasters.com) online service (optional)
^^ Default graphics preset is **High**

**Comment by hswhite33** [comment]
Had a quick play around - seems to work well but performance is noticeably worse than in Windows (framerate dropping to low-mid 40s).

Logitech F310 gamepad works without any extra configuration, but the game doesn't detect Logitech G29 racing wheel.

[System info](https://gist.github.com/hswhite33/d5872273b13aa91e29a7104c1c4866a9)

Proton 3.7-5 Beta

**Comment by theMarix** [comment]
@hswhite33 , have you tried using the G29 in the G27 mode? AFAIK Rally also only works in that mode.

**Comment by hswhite33** [comment]
@theMarix yes i tried in both g27 and g29 modes; the wheel was not detected in either mode.

**Comment by theMarix** [comment]
@hswhite33 Thanks for the clarification!

**Comment by Zero86Sk** [comment]
the game runs fine on proton 3.7-7 beta

there is a bit of a performance hit but i can still play it on 1050ti on high at 45 - 50 fps

i cant connect to racenet but multiplayer sessions seem to work

**Comment by SteveHeist** [comment]
Game runs, but after a certain "Complexity" of the generated map, it just doesn't load the level.

**Comment by theMarix** [comment]
For me it also gets stuck on loading the stage. But this even happens e.g. in Wales on minimum complexity and stage length.

**Comment by hswhite33** [comment]
Retested with Proton 3.16-1

G29 wheel now works, but no force feedback both in G29 mode and G27 mode. Performance is still sub-par

[System info](https://gist.github.com/hswhite33/813c9512b7bdd28d3ca414bfcf395e23)

**Comment by undu** [comment]
Retested with Proton 3.16-1 as well, it works fine.


Edit: remember to enable steam overlay for good controller support!

**Comment by Macribit** [comment]
any update on this?? for me racenet still doesn't work . But performance is really good I get over 60fps at medium setting with a rx460 which is an entry level gpu. This is at 1080p with Mesa 18.3 rc4, stable frames.
Proton 3.7-8beta.

**Comment by dragonfly-lover** [comment]
I have a good performance (30 to 50 FPS) on a 750ti. The game runs stable. Nevertheless i can't get Force feedback support in my Logitech driving Force ex, and can't connect to racenet. FF support in wine 3.16 seems to Be broken while i tried "wine control" on 3.20 staging and It works properly.

**Comment by Macribit** [comment]
now it's stuck on the loading stage screen. On Wine 3.20 it works Ok, weird maybe a game bug?? I played two hours with proton, now it refuses to pass the loading stage.

**Comment by alasky17** [comment]
@dragonfly-lover Did you have FF with Proton 3.7?  Also, could you check to make sure that "Xbox configuration support" is **not** turned on in the general Steam controller settings?  (That option is known-broken with Proton.)

**Comment by kisak-valve** [comment]
DiRT 4 doesn't start

Issue transferred from https://github.com/ValveSoftware/Proton/issues/2063.
@5005007 posted on 2018-12-06T21:27:38:

# Compatibility Report
- Name of the game with compatibility issues: DiRT 4
- Steam AppID of the game: 421020

## System Information
- GPU: 1060
- Driver/LLVM version: 410.78
- Kernel version: 4.15.0-42-generic
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/5005007/2a276b2cdce17be5a419d268f4001c97
- Proton version: 3.7-8

## I confirm:
- [*] that I haven't found an existing compatibility report for this game.
- [*] that I have checked whether there are updates for my system available.


[steam-421020.log](https://github.com/ValveSoftware/Proton/files/2654974/steam-421020.log)

## Symptoms 
Game doen't start

## Reproduction

**Comment by kisak-valve** [comment]
Hello @5005007, `err:   DxvkInstance: Failed to create instance` `err:   D3D11CreateDevice: Failed to create a DXGI factory` in the log hints that your vulkan driver may not be working properly, does any vulkan application work like `vulkaninfo` from `vulkan-utils`?

**Comment by 5005007** [comment]
Hello @kisak-valve 
I've got 

VULKANINFO

Vulkan Instance Version: 1.1.92

Cannot create Vulkan instance.
/build/vulkan-tools-1.1.92.0~rc3/vulkaninfo/vulkaninfo.c:835: failed with VK_ERROR_INCOMPATIBLE_DRIVER

**Comment by kisak-valve** [comment]
It's odd that the instance version would be that new for ubuntu 18.04 and nvidia 410.78, but regardless, that's a pretty strong indicator that you have a video driver issue that needs to be fixed and it's not a game-specific issue.

**Comment by 5005007** [comment]
I've just created a file 
/usr/share/vulkan/icd.d/nvidia_icd.json
```
{
    "file_format_version" : "1.0.0",
    "ICD": {
        "library_path": "libGLX_nvidia.so.0",
        "api_version" : "1.1.82"
    }
}
```

And everything works like a charm! Thank you very much!
The issue was solved.

**Comment by dragonfly-lover** [comment]
@alasky17 Nope. FF didn't work also in Proton 3.7. this problem happens also with gas guzzlers Extreme. When i play with "wine control" with proton's wine i see that FF doesn't work properly. In wine 3.20 the joystick calibration tool seems to work properly. So It should Be a matter of waiting the next major release of Proton.

**Comment by seansplayin** [comment]
Dirt 4 game performance rocks on AMD Vega 64 under linux and if there is any performance delta with  Windows 10 I sure can't tell.
currently using Ubuntu 18.10. on kernel 4.20 with mesa 18.3 and proton 3.16beta but also ran good on ubuntu 18.04 on kernel 4.18 with mesa 18.05 and proton 3.7.
just ran test, freeplay land rush Nevada championship with graphics settings on "high" and using a 3440x1440p monitor average likely around 95fps, lowest I saw on screen was 64fps and highest was 115fps.  freeplay land rush California is likely over 100fps average.

I'm also running AMD Ryzen 2700x @ 4.3ghz all core/4.35 4 core boost with 3600 memory at 14 timings. Kernel optimizations for AMD Ryzen are obvious, game runs on 16 threads with most thread very balanced, it's highly possible if you have only 4-6 cpu cores your experience may not be as good as mine, running a windows game through a compatibility layer is going to require some additional cpu resources.
only complaint is can't connect to racenet to sync results.

**Comment by harogaston** [comment]
Game was working fine until last time I tried (maybe two months ago). Now, several Steam App, vulkan, kernel and nvidia-drivers updates later it does not even launch.
Here is log output, I see some ERROR messages but couldn't figure out what to do.

[steam.log](https://github.com/ValveSoftware/Proton/files/2872483/steam.log)

**Comment by kisak-valve** [comment]
Hello @harogaston, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report. Also, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and drag and drop the generated $HOME/steam-$APPID.log into the comment box.

**Comment by harogaston** [comment]
@kisak-valve 
System information: [gist](https://gist.github.com/harogaston/87968f34d92b59cebc7156308006e7de)
Steam app log: [steam-421020.log](https://github.com/ValveSoftware/Proton/files/2872498/steam-421020.log)

**Comment by Macribit** [comment]
@harogaston 

same happens to me emulate virtual desktop in winecfg fixes it.

but unfortunately with 3.16-7beta sound is choppy, other versions work though.

**Comment by harogaston** [comment]
@Macribit Can you tell me where do I find winecfg and what to edit?
